### PR TITLE
Easier custom columns with drake_plan(), plus a new bind_rows() function

### PR DESCRIPTION
**Pull request by wlandau**
# Summary

Users can now use a `target()` pseudo-function in commands to customize the columns of `drake_plan()`. `target() is not an actual function, which is fine because calls to `target()` get preprocessed away before commands are executed.

```r
library(drake)
drake_plan(
  website_data = target(
    command = download_data("www.your_url.com"), 
    trigger = "always",
    custom_column = 5
  ),
  analysis = analyze(website_data), 
  strings_in_dots = "literals"
)
#> # A tibble: 2 x 4
#>   target       command                               trigger custom_column
#>   <chr>        <chr>                                 <chr>           <dbl>
#> 1 website_data "download_data(\"www.your_url.com\")" always             5.
#> 2 analysis     analyze(website_data)                 any               NA
```

There is also a new `bind_plans()` function to concatenate the rows of different plans and then sanitize the result.

```r
library(drake)
# You might need to refresh your data regularly (see ?triggers).
download_plan <- drake_plan(
  data = target(
    command = download_data(),
    trigger = "always"
  ), 
  strings_in_dots = "literals"
)
# But if the data don't change, the analyses don't need to change.
analysis_plan <- drake_plan(
  usage = get_usage_metrics(data),
  topline = scrape_topline_table(data)
)
your_plan <- bind_plans(download_plan, analysis_plan)
your_plan
#> # A tibble: 3 x 3
#>   target  command                    trigger
#>   <chr>   <chr>                      <chr>  
#> 1 data    download_data()            always 
#> 2 usage   get_usage_metrics(data)    any    
#> 3 topline scrape_topline_table(data) any
```

This PR also adds examples to the help files, the ["packages" example](https://github.com/ropensci/drake/tree/master/inst/examples/packages), and ["packages" vignette](https://github.com/ropensci/drake/blob/master/vignettes/example-packages.Rmd).

# GitHub issues fixed

- Ref: #315 

# Checklist

- [x] I have read `drake`'s [code of conduct](https://github.com/ropensci/drake/blob/master/CONDUCT.md), and I agree to follow its rules.
- [x] I have read the [guidelines for contributing](https://github.com/ropensci/drake/blob/master/CONTRIBUTING.md).
- [x] I have listed any substantial changes in the [development news](https://github.com/ropensci/drake/blob/master/NEWS.md).
- [x] I have added [`testthat`](https://github.com/r-lib/testthat) unit tests to [`tests/testthat`](https://github.com/ropensci/drake/tree/master/tests/testthat) to confirm that any new features or functionality work correctly.
- [x] I have tested this pull request locally with `devtools::check()`
- [x] This pull request is ready for review.
- [x] I think this pull request is ready to merge.

cc @krlmlr, @AlexAxthelm, @rkrug

### CODE REVIEW

File: R/sanitize.R [hunk 1]
```diff
@@ -10,12 +10,18 @@ sanitize_plan <- function(plan){
...
-  if (!is.null(plan[["trigger"]])){
+  if ("trigger" %in% colnames(plan)){
+    plan$trigger[is.na(plan$trigger) | !nzchar(plan$trigger)] <- "any"
     assert_legal_triggers(plan[["trigger"]])
   }
   plan <- file_outs_to_targets(plan)
   plan$target <- repair_target_names(plan$target)
-  plan[nzchar(plan$target), ]
+  plan <- plan[nzchar(plan$target), ]
+  plan[["seq_len(n())"]] <- NULL
```
Comment (krlmlr) [inline R/sanitize.R]: Where does that sentinel come from?

File: R/workplan.R [hunk 2]
```diff
@@ -103,6 +103,17 @@
 #' # There, `drake` sees that `small`, `large`, and `coef_regression2_small`
 #' # are loaded in with calls to `loadd()` and `readd()`.
 #' deps("report.Rmd")
+#' # You can create your own custom columns too.
+#' # See ?triggers for more on triggers.
+#' drake_plan(
+#'   website_data = target(
```
Comment (krlmlr) [inline R/workplan.R]: I like the interface. Would you like to export a function by that name?

File: R/dependencies.R [hunk 3]
```diff
@@ -500,3 +515,10 @@ is_readd_call <- function(expr){
 is_ignore_call <- function(expr){
   wide_deparse(expr[[1]]) %in% ignore_fns
 }
+
+is_target_call <- function(expr){
+  tryCatch(
+    wide_deparse(expr[[1]]) %in% target_fns,
```
Comment (krlmlr) [inline R/dependencies.R]: I wonder if tidy evaluation could help make this a bit more elegant.

File: R/workplan.R [hunk 4]
```diff
@@ -103,6 +103,17 @@
...
+#'     command = download_data("www.your_url.com"),
```
Comment (krlmlr) [inline R/workplan.R]: Do we need the named `command` argument?

File: R/sanitize.R (hunk 1 above)
Comment (wlandau) [reply R/sanitize.R]: You mean `plan[["seq_len(n())"]]`? `parse_custom_columns()` (in `workplan.R`) created that superfluous column in a `group_by() %>% do()`. I suppose `out[["seq_len(n())"]] <- NULL` belongs there. See 04c6dfc40b812d9240f8665a27b0a1bebc486895.

File: R/dependencies.R (hunk 3 above)
Comment (wlandau) [reply R/dependencies.R]: How so? I do want to adopt tidy evaluation throughout `drake`.

File: R/workplan.R (hunk 4 above)
Comment (wlandau) [reply R/workplan.R]: Currently yes, but we can change that.

File: R/workplan.R (hunk 2 above)
Comment (wlandau) [reply R/workplan.R]: Good idea. I am mildly concerned that `target()` could conflict with other names, but I think this is unlikely.

File: R/dependencies.R (hunk 3 above)
Comment (krlmlr) [reply R/dependencies.R]: If `expr` is a quosure, and we evaluate `expr[[1]]` in the quosure's environment, we can compare the result directly with `drake::target` and don't need to go via the textual representation. Maybe later.

Comment (wlandau) [reply R/dependencies.R]: If we do, it would be great to drop the textual representation in all `is_*_call()` functions.

**Pull request merged**